pkg/utils/utils.go

```diff
@@ -29,3 +32,7 @@ func BuildKubeClient(kubeConfigPath string) (*kubernetes.Clientset, error) {
 func ResourceNamespacedName(resourceType, namespace, name string) string {
 	return fmt.Sprintf("%s/%s/%s", resourceType, namespace, name)
 }
+
+func ClusterIsOffLine(conditions []metav1.Condition) bool {
+	return !meta.IsStatusConditionTrue(conditions, clusterapiv1.ManagedClusterConditionAvailable)
```

Comment:
we should  only consider unknown  status

Reply:
I found that we have false status for ManagedClusterConditionAvailable when managed cluster api server is Unavailable. https://github.com/open-cluster-management-io/registration/blob/e6435f32a3ed877b974f3299bd26db1101162f7b/pkg/spoke/managedcluster/healthcheck_controller.go#L96

Reply:
done